### PR TITLE
Fix #12346: Spinner respect Min/Decimal annotations

### LINKED ISSUE #12346: Spinner: Not respecting @Min or @DecimalMin correctly
### Describe the bug

Spinner is not correctly finding `@Min(0)` or `@DecimalMin`

```xml
<p:spinner id="basic" value="#{spinnerView.number1}" required="true">
```

```java
    @Min(0)
    @Max(100)
    private Integer number1;
```

### Reproducer

[pf-spinner.zip](https://github.com/user-attachments/files/16418953/pf-spinner.zip)

### Expected behavior

Min should set the component minimum.

### PrimeFaces edition

Community

### PrimeFaces version

14.0.3

### Theme

ALL

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #12346: Spinner respect Min/Decimal annotations

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/metadata/transformer/AbstractInputMetadataTransformer.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ else if (input instanceof InputTextarea) {
         return Integer.MIN_VALUE;
     }
 
-    protected boolean isMaxlenghtSet(UIInput input) {
+    protected boolean isMaxLengthSet(UIInput input) {
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/metadata/transformer/AbstractInputMetadataTransformer.java]: It was by design, see the getter or HTML attr name

Comment (melloware) [reply primefaces/src/main/java/org/primefaces/metadata/transformer/AbstractInputMetadataTransformer.java]: OK i changed the L to lowercase but the word "length" was misspelled "lenght"

**Comment by tandraschko** [comment]
Maybe we should create getter on the spinner for the Default value and reuse there and in the StateHelper
Ofherwise it will get out of sync again

**Comment by melloware** [comment]
good idea.

**Pull request merged**